### PR TITLE
Add auth integration tests

### PR DESCRIPTION
**Pull request by woop**
**What this PR does / why we need it**:

Adds authentication and authorization integration tests. These tests use actual backing services like Postgres, Kafka, Keto, and a fake JWKs server to simulate a real deployment. 

Will be updated and merged once #886 is merged in.

**Which issue(s) this PR fixes**:

Partly addresses #860, but not enough to close.

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: auth/src/main/java/feast/auth/config/SecurityConfig.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ GrpcAuthenticationReader authenticationReader() {
    * @return AccessDecisionManager
    */
   @Bean
-  @ConditionalOnProperty(prefix = "feast.security.authorization", name = "enabled")
+  @ConditionalOnProperty(prefix = "feast.security.authentication", name = "enabled")
```
Comment (jmelinav) [inline auth/src/main/java/feast/auth/config/SecurityConfig.java:92]: We might have to revert this change till #865 gets merged in. As you already know, If any deployment has authentication enabled but not authorization, serving boot up will fail due to connection dependency on core.

Comment (woop) [reply auth/src/main/java/feast/auth/config/SecurityConfig.java:92]: Correct.

File: core/src/test/java/feast/core/auth/CoreServiceAuthorizationIT.java [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    Channel insecureChannel =
+        ManagedChannelBuilder.forAddress("localhost", feast_core_port).usePlaintext().build();
+    insecureCoreService = CoreServiceGrpc.newBlockingStub(insecureChannel);
+  }
+
+  @AfterAll
+  static void tearDown() {
+    environment.stop();
+    wireMockRule.stop();
+    CollectorRegistry.defaultRegistry.clear();
```
Comment (pyalex) [inline core/src/test/java/feast/core/auth/CoreServiceAuthorizationIT.java:157]: this is being called in BaseIT.tearDown, could be omitted here

File: core/pom.xml [hunk 3]
```diff
@@ -315,12 +314,24 @@
...
+            <version>5.6.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-params</artifactId>
+            <version>5.6.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (pyalex) [inline core/pom.xml:329]: junit-jupiter-engine is being installed as failsafe dependency to prevent conflicts with junit-vintage-engine, so not sure it needed here, but on the other hand everything works
